### ISSUE TITLE
Soldiers cannot cross narrow water/swamp

### ISSUE THREAD

**Issue by fritzmg** [Return-To-The-Roots/s25client#1222]
Even though you can build across narrow water - it appears soldiers (or really anyone?) is unable to actually cross narrow water/swamp. This means it is impossible to attack any sort of military building, that is built somewhere which is only accessible through narrow water or swamp.

The lower left starting position on the map "Weihnachtsinsel" for example is completely inaccessible from the outside then (without catapults that is).

![Annotation 2020-05-12 193304](https://user-images.githubusercontent.com/4970961/81732139-eadbf180-9487-11ea-9c2c-6224db534c59.png)

`v0.8.2-0383c0c`

**Comment by Spikeone** [comment]
I'm not really sure how to argue, this is actually the correct behavior. 

On some maps this is used to control the gameplay, on this one it is just... well a funny gimmick which is annyoing.

I'd agree, that one may have an addon for that.

**Comment by fritzmg** [comment]
Hm, are you sure about that? Because I just noticed that sometimes soldiers _can_ cross narrow water and attack a military building, that is only accessible across narrow water.

**Comment by Spikeone** [comment]
Just tested it to make sure,

![image](https://user-images.githubusercontent.com/1276342/82147263-dd8b8200-984e-11ea-8f0a-65d0b704a797.png)

I'm not sure if they could pass those if this is simply the shortest route and there is another, but in s2gold it wasn't possible to attack across such "rivers"

**Comment by Flow86** [comment]
I thought it was possible as soon as one road went across?

**Comment by fritzmg** [comment]
> I thought it was possible as soon as one road went across?

Yeah, I think that's what happened in the previous game where I noticed this. Unfortunately I didn't take a screenshot or save a replay. But I'll have a closer look next time ;).

**Comment by Spikeone** [comment]
Yep, its only possible if there is a road - same as the original.

But this can result in quite strange situations. You need to wait for the enemy to build a road across a river... which he might not. Also you can build a road, attack, destroy the road and wait for soldiers to occupy your building.
So it's worth an addon in my opinion. Although there is only one map I know where this is a problem, maybe on pangea as well which makes two.

**Comment by q9f** [comment]
The same applies to animals/hunters, etc.

Sometimes you run into edge cases where you cannot build a road and the game is getting stuck (and the enemy won't build a road for you in most cases).
